common/test/acceptance/tests/lms/test_account_settings.py

```diff
@@ -110,15 +123,16 @@ def test_page_view_event(self):
         And I visit my account settings page
         Then a page view analytics event should be recorded
         """
-        self.verify_events_of_type(
-            self.username,
-            u"edx.user.settings.viewed",
-            [{
-                u"user_id": long(self.user_id),
-                u"page": u"account",
-                u"visibility": None,
-            }]
-        )
+
+        event_filter = {
+            'event_type':'edx.user.settings.viewed',
+            'event': {
+                'user_id': self.user_id,
+                'page': 'account',
+                'visibility': None
+            }
+        }
+        self.wait_for_events(event_filter=event_filter, number_of_matches=1)
```

Comment:
There's a subtle distinction that I'm still trying to think through. The old logic collected all ``edx.user.settings.viewed`` events, and then compared them against the expected set. The new logic waits for exactly the expected event and times out if one doesn't show up. If I'm reading this correctly, the failure message won't show the set of events that were actually returned and the diffs with the expected events. Instead it will just show a promise timeout. 

I really like that your new logic verifies the entire event, and not just the sub 'event' field. I just want to be sure that the diff I get back on failure really shows me why it failed. For example, were there two events instead of one, was it the wrong event type, was it just flat out missing?